packages/tests/server/react/useContext.test.tsx

```diff
@@ -798,4 +799,119 @@ describe('query keys are stored separtely', () => {
+    test('no input', async () => {
+      const { proxy, App } = ctx;
+
+      function MyComponent() {
+        const utils = proxy.useContext();
+        const happy = utils.post.all.getQueryKey();
+
+        // @ts-expect-error - post.all has no input
+        const sad = utils.post.all.getQueryKey('foo');
+
+        return <pre data-testid="qKey">{JSON.stringify(happy)}</pre>;
+      }
+
+      const utils = render(
+        <App>
+          <MyComponent />
+        </App>,
+      );
+
+      await waitFor(() => {
+        expect(utils.getByTestId('qKey')).toHaveTextContent(
+          JSON.stringify([['post', 'all'], { type: 'query' }]),
+        );
+      });
+    });
+
+    test('with input', async () => {
+      const { proxy, App } = ctx;
+
+      function MyComponent() {
+        const utils = proxy.useContext();
+        const happy = utils.post.byId.getQueryKey({ id: 1 });
+
+        // @ts-expect-error - post.byId has required input
+        const sad1 = utils.post.byId.getQueryKey();
+        // @ts-expect-error - id should be a number
+        const sad2 = utils.post.byId.getQueryKey({ id: '1' });
+
+        return <pre data-testid="qKey">{JSON.stringify(happy)}</pre>;
+      }
+
+      const utils = render(
+        <App>
+          <MyComponent />
+        </App>,
+      );
+
+      await waitFor(() => {
+        expect(utils.getByTestId('qKey')).toHaveTextContent(
+          JSON.stringify([
+            ['post', 'byId'],
+            { input: { id: 1 }, type: 'query' },
+          ]),
+        );
+      });
+    });
+
+    test('on router', async () => {
+      const { proxy, App } = ctx;
+
+      function MyComponent() {
+        const utils = proxy.useContext();
+        const happy = utils.post.getQueryKey();
+
+        // @ts-expect-error - router has no input
+        const sad = utils.post.getQueryKey('foo');
+
+        return (
+          <div>
+            <pre data-testid="qKey">{JSON.stringify(happy)}</pre>
+          </div>
+        );
+      }
+
+      const utils = render(
+        <App>
+          <MyComponent />
+        </App>,
+      );
+
+      await waitFor(() => {
+        expect(utils.getByTestId('qKey')).toHaveTextContent(
+          JSON.stringify([['post']]),
+        );
+      });
+    });
+
+    test('forwarded to a real method', async () => {
+      const { proxy, App } = ctx;
+
+      function MyComponent() {
+        const utils = proxy.useContext();
+        const qc = useQueryClient();
+
+        const key = utils.post.all.getQueryKey();
+        const isFetching = qc.isFetching({ queryKey: key });
+
+        return <pre data-testid="isFetching">{isFetching}</pre>;
+      }
+
+      const utils = render(
+        <App>
+          <MyComponent />
+        </App>,
+      );
+
+      // should be fetching initially, and then not
+      expect(utils.container).toHaveTextContent('1');
+      await waitFor(() => {
+        expect(utils.container).toHaveTextContent('0');
+      });
```

Comment:
@TkDodo This should pass, right?

Reply:
no, because you don't have a subscription. `qc.isFetching` just reads the value once. You'd want:

```
const isFetching = useIsFetching({ queryKey: key });
```

you can import it directly:

```
import { useIsFetching } from '@tanstack/react-query'
```

Reply:
oh right.. thanks for clarifying 🙏🏼